lib/ruby_smb/client/negotiation.rb

```diff
@@ -66,6 +65,11 @@ def negotiate_response(raw_data)
         response = nil
         if smb1
           packet = RubySMB::SMB1::Packet::NegotiateResponseExtended.read raw_data
+
+          unless packet.valid?
+            packet = RubySMB::SMB1::Packet::NegotiateResponse.read raw_data
+          end
```

Comment:
Just confirming if this is secure to do. I don't know enough about SMB, but if we've negotiated some sort of extended security session, it feels like falling back here feels like it should be predicated on the whether or not extended security was enabled in the first place? 👀

Reply:
Agreed, but that may be outside the scope of this PR.

Reply:
Again, I don't know the scope of this change - but if the PR adds a potential security regression, then it's definitely in the scope of this PR 😄 

cc @cdelafuente-r7 I assume there's no negative repercussions to this line? 👀

Reply:
The client needs to follow what the server claims to support. At this point, the client has sent a `NegotiateRequest` asking for [Extended Security](https://github.com/rapid7/ruby_smb/blob/master/lib/ruby_smb/client/negotiation.rb#L213) and the server should answer with the `CAP_EXTENDED_SECURITY` capability flag, meaning it supports Extended Security and the response contains the expected Security Blob (GSS protocol negotiation). Parsing first with a `NegotiateResponseExtended` structure will just do that. If the `CAP_EXTENDED_SECURITY` capability flag is not set, `packet.valid?` will fail and the client will fail back to a normal `NegotiateResponse`. This process is described [here](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb/d367854f-5eee-45e8-a588-eed596a1a521).

That said, The Security Blob is not used this way, for simplicity. RubySMB assumes the server uses NTLMSSP as authentication mechanism and generates the corresponding Security Blob in the upcoming Session Setup request, which follows the negotiation step. It ignores the `NegotiateResponseExtended` Security Blob (GSS protocol negotiation). RubySMB does not support other authentication mechanism anyway. So, if the server does not support NTLMSSP or any Extended Security related protocol at all, the authentication is likely to fail.